flight/PiOS/STM32F4xx/pios_flash_internal.c

```diff
@@ -34,6 +36,9 @@
 #include "pios_semaphore.h"
 #include <stdbool.h>
 
+/* Todo: consider relocating to a good header */
+bool __attribute__((weak)) vehicle_is_armed = false;
```

Comment:
Yes? What I'm proposing is making this a static local var, and adding a public method to set it which the module can call.